src/main/java/org/jnode/fs/xfs/inode/INode.java

```diff
@@ -80,6 +150,24 @@ public int getMode() {
         return getUInt16(0x2);
     }
 
+    /**
+     * Is Directory the current inode.
+     *
+     * @return true.
+     */
+    public boolean isDirectory() throws IOException {
+        return FileMode.is((int) getMode(),FileMode.DIRECTORY);
+    }
+
+    /**
+     * Gets the mode stored in this inode.
+     *
+     * @return the mode.
+     */
+    public boolean isFile() throws IOException {
+        return FileMode.is((int) getMode(),FileMode.FILE);
```

Comment:
```suggestion
     * Returns {@code true} if the current inode is a file.
     *
     * @return {@code true} if the current inode is a file, {@code false} otherwise.
     */
    public boolean isFile() {
        return FileMode.is(getMode(), FileMode.FILE);
```

Reply:
The suggestion was applied, thanks Erwin.